### PR TITLE
Ban tokio sleep and tokio filesystem writes via semgrep

### PR DESCRIPTION
**Pull request by justinmoon**
semgrep doesn't work on mac so this I haven't tested this (cc @dpc)

### CODE REVIEW

File: .semgrep.all.yaml [hunk 1]
```diff
@@ -12,12 +12,23 @@ rules:
...
-  pattern: std::fs::File::create
+  pattern-either:
+    - pattern: std::fs::File::create
+    - pattern: tokio::fs::File::create
+  severity: WARNING
+
+- id: ban-tokio-sleep
+  languages:
+    - rust
+  message: `tokio::time::sleep` doesn't work in WASM. Use `fedimint_core::task::sleep` instead.
```
Comment (dpc) [inline .semgrep.all.yaml:32]: Need to put this whole in `".."`, methinks. Otherwise yaml gets spooked.

**Comment by dpc** [comment]
Semgrep does not display validation errors, reported upstream.

Check with `semgrep scan --config .semgrep.all.yaml`.

Fixes needed:

```diff
> git diff HEAD                                                                                      
diff --git a/.semgrep.all.yaml b/.semgrep.all.yaml                                                   
index bd2655fa4b..3e82c10766 100644               
--- a/.semgrep.all.yaml                                                                              
+++ b/.semgrep.all.yaml                                                                              
@@ -13,8 +13,8 @@ rules:                                                                             
     - rust                                                                                          
   message: Overwritting existing files is rarely a good idea. Use fedimint-core::util::write_overwrite if you are really sure you want to overwrite existing data or `fedimint-core::util::write_overwrite::write_new` if you don't. Alternatively use `fs::File::options` and express intention explicitly.   
   pattern-either:                                                                                   
-    - tokio::fs::write                                                                              
-    - std::fs::write                                                                                
+    - pattern: tokio::fs::write                                                                     
+    - pattern: std::fs::write                                                                       
   severity: WARNING                                                                                 
                                                                                                     
 - id: ban-file-create                                                                               
@@ -29,6 +29,6 @@ rules:                                                                             
 - id: ban-tokio-sleep                                                                               
   languages:                                                                                        
     - rust                                                                                          
-  message: `tokio::time::sleep` doesn't work in WASM. Use `fedimint_core::task::sleep` instead.     
+  message: "`tokio::time::sleep` doesn't work in WASM. Use `fedimint_core::task::sleep` instead."   
   pattern: tokio::time::sleep                                                                       
   severity: WARNING 
```

but unfortunately this reports bunch of existing instances.

**Comment by justinmoon** [comment]
converted to draft because of ^^. feel free to take this over @dpc because semgrep doesn't work on my machine so i can't test it.

**Comment by justinmoon** [comment]
@dpc could you try to finish this off? lots of problems with `tokio::time` recently ...

**Comment by dpc** [comment]
Moved to https://github.com/fedimint/fedimint/pull/2564

**Comment by justinmoon** [comment]
https://github.com/fedimint/fedimint/pull/2564 did this

**Pull request closed without merging**